File: ts/wam/6506.ts
export const WamForwardActionUserJourney = 6506 as const

export interface ForwardActionUserJourneyEvent {
  /** field 1, wire `app_session_id` */
  appSessionId?: string
  /** field 2, wire `forward_action_user_journey_action` */
  forwardActionUserJourneyAction?: FORWARDACTIONUSERJOURNEYACTION
  /** field 3, wire `message_is_from_me` */
  messageIsFromMe?: boolean
  /** field 4, wire `message_media_type` */
  messageMediaType?: MEDIATYPE
  /** field 5, wire `message_type` */
  messageType?: MESSAGETYPE
  /** field 6, wire `ui_surface` */
  uiSurface?: TSSURFACE
  /** field 7, wire `user_journey_chat_type` */
  userJourneyChatType?: USERJOURNEYCHATTYPE
  /** field 8, wire `user_journey_funnel_id` */
  userJourneyFunnelId?: string
  /** field 9, wire `unified_session_id` */
  unifiedSessionId?: string
  /** field 11, wire `forward_user_journey_funnel_id` */
  forwardUserJourneyFunnelId?: string
  /** field 12, wire `user_journey_event_ms` */
  userJourneyEventMs?: number
}

File: go/wam/6506.go
const WamForwardActionUserJourney = 6506 // channel: regular

type ForwardActionUserJourneyEvent struct {
	AppSessionId string `wam:"app_session_id"` // field 1
	ForwardActionUserJourneyAction FORWARDACTIONUSERJOURNEYACTION `wam:"forward_action_user_journey_action"` // field 2
	MessageIsFromMe bool `wam:"message_is_from_me"` // field 3
	MessageMediaType MEDIATYPE `wam:"message_media_type"` // field 4
	MessageType MESSAGETYPE `wam:"message_type"` // field 5
	UiSurface TSSURFACE `wam:"ui_surface"` // field 6
	UserJourneyChatType USERJOURNEYCHATTYPE `wam:"user_journey_chat_type"` // field 7
	UserJourneyFunnelId string `wam:"user_journey_funnel_id"` // field 8
	UnifiedSessionId string `wam:"unified_session_id"` // field 9
	ForwardUserJourneyFunnelId string `wam:"forward_user_journey_funnel_id"` // field 11
	UserJourneyEventMs int64 `wam:"user_journey_event_ms"` // field 12
}